frontend/mina-signer/src/MinaSDK.ts

```diff
@@ -0,0 +1,356 @@
+"use strict";
+
+const JSOfOCaml_SDK = require("./client_sdk.bc.js");
+const minaSDK = JSOfOCaml_SDK.minaSDK;
+
+import {
+  network,
+  publicKey,
+  keypair,
+  privateKey,
+  signed,
+  payment,
+  stakeDelegation,
+  message,
+} from "./TSTypes";
+
+const defaultValidUntil = "4294967295";
+
+class Client {
+  private network: network;
+
+  constructor(options: { network: network }) {
+    if (options.network !== "mainnet" && options.network !== "testnet") {
+      throw "Invalid Specified Network";
+    }
+    this.network = options.network;
+  }
+
+  /**
+   * Generates a public/private keypair
+   *
+   * @returns A Mina keypair
+   */
+  public genKeys(): keypair {
+    return minaSDK.genKeys();
+  }
+
+  /**
+   * Verifies if a keypair is valid by checking if the public key can be derived from
+   * the private key and additionally checking if we can use the private key to
+   * sign a transaction. If the keypair is invalid, an exception is thrown.
+   *
+   * @param keypair - A keypair
+   * @returns True if the `keypair` is a verifiable keypair, otherwise throw an exception
+   */
+  public verifyKeypair(keypair: keypair): boolean {
+    return minaSDK.validKeypair(keypair);
+  }
+
+  /**
+   * Derives the public key of the corresponding private key
+   *
+   * @param privateKey - The private key used to get the corresponding public key
+   * @returns A public key
+   */
+  public derivePublicKey(privateKey: publicKey): publicKey {
+    return minaSDK.publicKeyOfPrivateKey(privateKey);
+  }
+
+  /**
+   * Signs an arbitrary message
+   *
+   * @param message - An arbitrary string message to be signed
+   * @param key- The keypair used to sign the message
+   * @returns A signed message
+   */
+  public signMessage(message: string, key: keypair): signed<message> {
+    return {
+      signature: minaSDK.signString(this.network, key.privateKey, message),
+      data: {
+        publicKey: key.publicKey,
+        message,
+      },
+    };
+  }
+
+  /**
+   * Verifies that a signature matches a message.
+   *
+   * @param signedMessage - A signed message
+   * @returns True if the `signedMessage` contains a valid signature matching
+   * the message and publicKey.
+   */
+  public verifyMessage(signedMessage: signed<message>): boolean {
+    return minaSDK.verifyStringSignature(
+      this.network,
+      signedMessage.signature,
+      signedMessage.data.publicKey,
+      signedMessage.data
+    );
+  }
+
+  /**
+   * Signs a payment transaction using a private key.
+   *
+   * This type of transaction allows a user to transfer funds from one account
+   * to another over the network.
+   *
+   * @param payment - An object describing the payment
+   * @param privateKey- The private key used to sign the transaction
+   * @returns A signed payment transaction
+   */
+  public signPayment(
+    payment: payment,
+    privateKey: privateKey
+  ): signed<payment> {
+    const memo = payment.memo ? payment.memo : "";
```

Comment:
A cool new way to write this is
```js
const memo = payment.memo ?? "";
```

Reply:
ah, that's neat! I changed the ternary statements to use this now, thanks! :D